Documentation

galfert opened the pull request:
Continuation of #1011 but without the the refactorings.

Refs #1005

gregkare commented on src/remotestorage.js:
```diff
@@ -181,16 +181,15 @@ RemoteStorage.prototype = {
 
   /**
-   * TODO: where is this defined?
+   * @property {object} remote
```
The `remote` property is still also documented on line 800 in that file:

```
Access to the remote backend used. Usually a <RemoteStorage.WireClient>.
```

galfert replied:
I removed the duplicate.

gregkare commented:
Is this one still WIP? I found a small issue but I think we could merge this one

skddc commented:
Yes.

skddc commented on package.json:
```diff
@@ -17,7 +17,8 @@
     "postversion": "git push && git push --tags",
-    "update-bower-version": "bash scripts/update-bower-version.sh"
+    "update-bower-version": "bash scripts/update-bower-version.sh",
+    "autobuild-docs": "cd doc && sphinx-autobuild . _build/html; cd .."
```
The way @gregkare explained it, we don't need to CD back out here, because the script is run in a new shell in a subprocess iirc.

gregkare replied:
My comment was about the Makefile, I don't know how npm run-script behaves

gregkare replied:
I just tried it, going back isn't needed

galfert commented:
Ok, I split up the legacy "Caching" doc and moved the parts to where (I think) they make sense.
I introduced an "Internals" section under "Contributing" where I moved the "Discovery bootstrap", parts of "Caching" and the "Data format of the local cache". I completely removed the legacy "AMD" document, as all the information is in the "Getting started" doc now. I also improved some of the JSDoc along the way and added a script for `npm run autobuild-docs`.

This is ready for review now.

skddc commented on src/remotestorage.js:
```diff
@@ -798,12 +798,6 @@ Object.defineProperty(RemoteStorage.prototype, 'caching', {
 /*
- * @property remote
- *
- * Access to the remote backend used. Usually a <RemoteStorage.WireClient>.
```
Has this property gone, or is it just `@private`?

galfert replied:
This was just a duplicate. I documented the property in https://github.com/remotestorage/remotestorage.js/pull/1037/files#diff-a18ce18f9db57ab2aeeb4b2e649342eaL184

skddc replied:
Cool.

skddc commented on doc/contributing/internals.rst:
```diff
@@ -0,0 +1,11 @@
+=========
+
+This section contains information about some of the internals and
+concepts of the remoteStorage library.
```
remoteStorage**.js**

skddc commented on doc/contributing/internals.rst:
```diff
@@ -0,0 +1,11 @@
+.. toctree::
+   :maxdepth: 2
+
+   internals/discovery-bootstrap
```
I still don't understand that document at all. It doesn't explain what's being explained, and even I as a core developer just have no idea what it's talking about really.

gregkare replied:
I also don't understand the Discovery bootstrap document. I don't know what "bootstrap" means in this case, it's only used in the title

gregkare commented on doc/contributing/internals/caching.rst:
```diff
@@ -0,0 +1,25 @@
+
+1. Setting caching strategy 'ALL' for a path creates an empty node for
+   that path, unless it already exists.
+2. The sync process will then do a 'GET', and create new nodes under any
```
I think this should be "The sync process will then do a GET request", at first I thought it was the same as 'ALL' above because of the quotes

gregkare commented on doc/js-api/caching.rst:
```diff
@@ -22,6 +43,19 @@ For each subtree, you can set the caching strategy to ``ALL``, ``SEEN``
+
+.. code:: javascript
+
+   remoteStorage.startSync().then(function() {
```
I can't get this to work, `remoteStorage.startSync()` always returns `undefined` (testing in my RS 1.0 branch of myfavoritedrinks)

gregkare replied:
Just remoteStorage.startSync() works, it doesn't return a promise like sync() used to do

skddc replied:
That was added after RC1 in 501031b1cb9e91f7ccccb0fcb58cd0c1366a6b95 (is part of RC2)

gregkare replied:
Thanks, I had forgotten to update to RC2. This works as advertised!

gregkare commented on src/baseclient.js:
```diff
@@ -299,12 +301,15 @@ BaseClient.prototype = {
-   * @param {string} path
-   * @param {string} strategy
+   * @param {string} path - Path to cache
+   * @param {string} strategy - Caching strategy. One of 'ALL', 'SEEN', or 'FLUSH'.
```
I would add that it is optional and defaults to 'ALL'

galfert commented:
I implemented your comments and tried to improve the discovery doc.

skddc commented:
@galfert Please use `docs/123-desc` for branches with documentation changes. Thanks!

skddc commented:
Rebased on master, edited some more, and merged locally.
